### ISSUE TITLE
How to add hidden layers

### ISSUE THREAD

**Issue by noneGMJ** [tkipf/gcn#44]
Hi,thanks for your work.
I have read your paper. Now I want to add a hidden layer to reproduce your result in appendix B.
So I do this :
 
        
        self.layers.append(GraphConvolution(input_dim=self.input_dim,
                                            output_dim=FLAGS.hidden1,
                                            placeholders=self.placeholders,
                                            act=tf.nn.relu,
                                            dropout=True,
                                            sparse_inputs=True,
                                            logging=self.logging))

        self.layers.append(GraphConvolution(input_dim=FLAGS.hidden1,
                                            output_dim=FLAGS.hidden1,
                                            placeholders=self.placeholders,
                                            act=lambda x: x,
                                            dropout=True,
                                            logging=self.logging))

        self.layers.append(GraphConvolution(input_dim=FLAGS.hidden1,
                                            output_dim=self.output_dim,
                                            placeholders=self.placeholders,
                                            act=lambda x: x,
                                            dropout=True,
                                            logging=self.logging))
        

And  hyperparameters I used are:

      'dataset'       : 'cora',
      'model'         : 'gcn',
      'learning_rate' : 0.01,
      'epochs'        : 400,
      'hidden1'      : 16,
      'dropout'       : 0.5,
      'weight_decay'  : 5e 4,
      'early_stopping': 0, 
   

But the version I modified only has 79.4% acc less than 90% in paper.
So what is my fault?
Thanks!

**Comment by tkipf** [comment]
We used 5-fold cross-validation for the results in the appendix (see description of Figure 5), so that gives you a lot more training data points than in the original split used in the rest of the paper. This should explain the higher accuracies in these figures. Hope this helps!

**Comment by guotong1988** [comment]
Where are the more training data？
